### PR TITLE
[scanning] Implement tab navigation for voucher entry flow (screens/scanning)

### PR DESCRIPTION
**Pull request by sauhardjain**
[//]: # "These comments are meant for your reference. They are invisible and don't need to be deleted!"

### Description
[//]: # "Required - Describe what's new in this PR in a few lines. A description and bullet points for specifics will suffice."
Implemented a tab-based navigation header to switch between the three voucher entry methods. 

- Built `VoucherEntryNavigator.tsx` to wrap the different voucher entry screens — conditionally rendering them through the tab header.
- Wrapped `VoucherEntryNavigator.tsx` in a `TouchableWithoutFeedback` that dismisses the keyboard whenever a user clicks anywhere on the screen.
- Disabled `autoFocus` on single digit `OTPTextInput` and add a Done button to the keyboard.
- Added a prop to `StandardHeader` to allow variable widths to be passed to it.

### Screenshots
[//]: # "Required for frontend changes, otherwise optional but strongly recommended. Add screenshots of expected behavior - GIFs if you're feeling fancy!"

IMG_3424
IMG_3425

## How to review
[//]: # "Required - Describe the order in which to review files and what to expect when testing locally. Is there anything specifically you want feedback on? Should this be reviewed commit by commit, or all at once? What are some user flows to test? What are some edge cases to look out for?"
Go through the Voucher Entry flow and test out all entry methods to see if they work as expected.

## Next steps
[//]: # "Optional - What's NOT in this PR, doesn't work yet, and/or still needs to be done. Note any temporary fixes in this PR that should be cleaned up later."

- [ ] Fix the typing error on `VoucherEntryNavigator`
- [x] Implement single-digit inputs on `VoucherBatchScreen`
- [x] Display a loading spinner on barcode screen before camera access is granted. (fixed by https://github.com/calblueprint/vouchers4veggies/pull/98)

### Related PRs
[//]: # "Optional - related PRs you're waiting on/ PRs that will conflict, etc; if this is a refactor, feel free to add PRs that previously modified this code"
[Tab Navigator from Sunday Friends](https://github.com/calblueprint/sunday-friends-mobile/pull/6) (s/o to the legends @CharlesMing02 @jacobjk01)

[//]: # "This tags the project leader as a default. Feel free to change, or add on anyone who you should be in on the conversation."
🥦 CC: @sauhardjain

### CODE REVIEW

File: src/components/common/StandardHeader.tsx [hunk 1]
```diff
@@ -3,13 +3,17 @@ import { View } from 'react-native';
 
 interface HeaderContainerProps {
   children: ReactNode;
+  width?: string;
 }
 
-export default function HeaderContainer({ children }: HeaderContainerProps) {
+export default function HeaderContainer({
+  children,
+  width,
```
Comment (selene-huang) [inline src/components/common/StandardHeader.tsx]: Nit: If you'd like width to default to 90%, you should change line 11 to `width = '90%'` and remove the `|| '90%'` in line 16

**Review by selene-huang — COMMENTED** [review]
Looks pretty good. A couple minor concerns:

- Between Serial and Range, the buttons shift upward slightly. I think it would be more intuitive if we had them aligned at the same height
- Every time I access barcode, the "No access to camera" displays briefly. Even if this is intended, there's like no styling on this message and it's allll the way at the top of my screen. If it's not intended - would it make more sense to replace it with a loading spinner?

File: src/screens/scanning/VoucherEntryNavigator.tsx
Comment (ronniebeggs) [inline src/screens/scanning/VoucherEntryNavigator.tsx:1]: Should this file be placed in the navigation folder?

File: src/screens/scanning/VoucherEntryNavigator.tsx [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+              case 1:
+                return <ManualVoucherScreen navigation={navigation} />;
+              case 2:
+                return <VoucherBatchScreen navigation={navigation} />;
+              case 3:
+                return <ScanningScreen navigation={navigation} />;
+              default:
+                return <ManualVoucherScreen navigation={navigation} />;
+            }
+          })()}
```
Comment (ronniebeggs) [inline src/screens/scanning/VoucherEntryNavigator.tsx:122]: figure out how to remove red squiggly lines

File: src/screens/scanning/VoucherEntryNavigator.tsx [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+              <NavButtonText
+                style={
+                  selection === 3
+                    ? { color: Colors.magenta }
+                    : { color: Colors.midGray }
+                }
+              >
+                Barcode
+              </NavButtonText>
+            </NavButton>
```
Comment (ronniebeggs) [inline src/screens/scanning/VoucherEntryNavigator.tsx:104]: Maybe make a NavButton component that takes in the button text as a prop. Removes repeated code.

**Review by ronniebeggs — APPROVED** [review]

File: src/screens/scanning/VoucherEntryNavigator.tsx
Comment (sauhardjain) [reply src/screens/scanning/VoucherEntryNavigator.tsx:1]: Since it's a screen, I think it's better to keep it in the existing folder.

File: src/screens/scanning/VoucherEntryNavigator.tsx (hunk 2 above)
Comment (sauhardjain) [reply src/screens/scanning/VoucherEntryNavigator.tsx:122]: To be fixed in a future PR
